rllib/utils/metrics/metrics_logger.py

```diff
@@ -376,256 +301,148 @@ def log_value(
-                        f"but got argument window={window} while the existing Stats object {key} "
-                        f"has window={stats._window}."
+                elif reduce == "lifetime_sum":
+                    self._set_key(
+                        key,
+                        LifetimeSumStats(
+                            _is_root_stats=self._is_root_logger,
+                            track_throughput_since_last_restore=with_throughput,
+                        ),
                     )
-                if percentiles != getattr(stats, "_percentiles", False) and log_once(
-                    f"percentiles_warning_{key}"
-                ):
-                    logger.warning(
-                        "percentiles should be the same for all logged values under the same key, "
-                        f"but got argument percentiles={percentiles} while the existing Stats object {key} "
-                        f"has percentiles={getattr(stats, '_percentiles', False)}."
+                elif reduce == "percentiles":
+                    self._set_key(
+                        key,
+                        PercentilesStats(
+                            _is_root_stats=self._is_root_logger,
+                            window=window,
+                            clear_on_reduce=clear_on_reduce,
+                            percentiles=percentiles,
+                        ),
                     )
-
-                if (
-                    reduce_per_index_on_aggregate
-                    != stats._reduce_per_index_on_aggregate
-                    and log_once(f"reduce_per_index_on_aggregate_warning_{key}")
-                ):
-                    logger.warning(
-                        f"reduce_per_index_on_aggregate should be the same for all logged values under the same key, "
-                        f"but got argument reduce_per_index_on_aggregate={reduce_per_index_on_aggregate} while the existing Stats object {key} "
-                        f"has reduce_per_index_on_aggregate={stats._reduce_per_index_on_aggregate}."
+                elif reduce == "item":
+                    assert window is None, "window is not supported for item reduction"
+                    self._set_key(
+                        key,
+                        ItemStats(
+                            _is_root_stats=self._is_root_logger,
+                            clear_on_reduce=clear_on_reduce,
+                        ),
+                    )
+                elif reduce == "item_series":
+                    self._set_key(
+                        key,
+                        ItemSeriesStats(
+                            _is_root_stats=self._is_root_logger,
+                            window=window,
+                            clear_on_reduce=clear_on_reduce,
+                        ),
+                    )
+                else:
+                    raise ValueError(
```

Comment:
How is custom reduce method handled, if custom `stats_cls_lookup` is passed?

Would user need to pass `reduce: str = "my_reduce",` to the `log_value()`?

Reply:
Yes, for now.
I hope we can make this interface a little nicer in the future.
For now, the main changes happen on the Stats classes side.